2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -4,15 +4,23 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# Used for wait_for_mongo_* functions
-MAX_ATTEMPTS=60
-SLEEP_TIME=1
-
+# Data directory where MongoDB database files live. The data subdirectory is here
+# because mongodb.conf lives in /var/lib/mongodb/ and we don't want a volume to
+# override it.
+export MONGODB_DATADIR=/var/lib/mongodb/data
 export MONGODB_CONFIG_PATH=/etc/mongod.conf
-export MONGODB_PID_FILE=/var/lib/mongodb/mongodb.pid
-export MONGODB_KEYFILE_PATH=/var/lib/mongodb/keyfile
+export MONGODB_KEYFILE_PATH="${HOME}/keyfile"
 export CONTAINER_PORT=27017
 
+# Configuration settings.
+export MONGODB_NOPREALLOC=${MONGODB_NOPREALLOC:-true}
+export MONGODB_SMALLFILES=${MONGODB_SMALLFILES:-true}
+export MONGODB_QUIET=${MONGODB_QUIET:-true}
+
+# Constants used for waiting
+export MAX_ATTEMPTS=60
+export SLEEP_TIME=1
```

Comment:
> Also MONGODB_CONFIG_PATH and MONGODB_KEYFILE_PATH, or not?

Here's where Bash haunts our souls again... :cry: 
I'm not sure if it is necessary to export in that case, I guess not necessary, need to try it out.

Here are the occurrences of MONGODB_CONFIG_PATH:

```console
$ git grep -n MONGODB_CONFIG_PATH upstream/master -- 2.4
upstream/master:2.4/root/usr/bin/run-mongod:67:envsubst < ${CONTAINER_SCRIPTS_PATH}/mongodb.conf.template > $MONGODB_CONFIG_PATH
upstream/master:2.4/root/usr/bin/run-mongod:71:mongo_common_args="-f $MONGODB_CONFIG_PATH --oplogSize 64"
upstream/master:2.4/root/usr/share/container-scripts/mongodb/common.sh:11:export MONGODB_CONFIG_PATH=/etc/mongod.conf
upstream/master:2.4/root/usr/share/container-scripts/mongodb/initiate_replica.sh:72:mongod -f ${MONGODB_CONFIG_PATH} --shutdown &>/dev/null
```

Trying it out:

```console
$ cat foo
#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

source bar

echo $V_EXPORTED
echo $V_NOT_EXPORTED
$ cat bar
#!/bin/bash

export V_EXPORTED="exported var"
V_NOT_EXPORTED="not exported var"
$ ./foo 
exported var
not exported var
```


Conclusion: no need to export MONGODB_CONFIG_PATH and MONGODB_KEYFILE_PATH, @omron93 agree?

BTW, MONGODB_KEYFILE_PATH is defined/exported both in `2.4/root/usr/bin/run-mongod:21` and `2.4/root/usr/share/container-scripts/mongodb/common.sh:13`, we should likely get rid of one of those lines. Notice that $HOME == /var/lib/mongodb.